### PR TITLE
fix(taglist): make tag remove accessible via keyboard

### LINKED ISSUE #413: Pills in the taglist component are not accessiblie
__Describe the Bug__

When using that taglist component if a user selects a value the pills of the selected values in the taglist component are not accessible.

image

__Steps to Reproduce__

1. Create a form with a taglist and render it with the form viewer
2. Select a value in the taglist
3. Now with keyboard only try to remove a value
4. Selected values can't be removed with keyboard only (no outline and pills are not clickable/focusable)

__Expected Behavior__

Values should be able to be removed with keyboard only and the pills should be focusable with an outline

__Environment__

- Host (Browser/Node version), if applicable: [e.g. MS Edge 18, Chrome 69, Node 10 LTS]
- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #413

This makes the tag remove pills actual buttons. I also added another a11y test to verify the focus state (dropdown list open).

Kapture 2022-11-22 at 14 18 34

### CODE REVIEW

File: packages/form-js-viewer/src/render/components/form-fields/Taglist.js [hunk 1]
```diff
@@ -106,6 +106,7 @@ export default function Taglist(props) {
       setIsEscapeClose(true);
       break;
     case 'Enter':
+      e.preventDefault();
```
[the remark this answers is not in the document]
Comment (vsgoulart) [reply packages/form-js-viewer/src/render/components/form-fields/Taglist.js:109]: Event bubbling maybe? But the event handler seems to not affect that

**Review by vsgoulart — COMMENTED** [review]
@pinussilvestrus I noticed that the focus path is a bit broken. When you have to focus on the list options and tab to the next element, if you tab back you have focus on the previous pill and not on the taglist input which open the available options.

Do you think we should fix that?

**Comment by pinussilvestrus** [comment]
Seems like the focus got lost after the removal of a tab. I'll have a look 👍

**Comment by pinussilvestrus** [comment]
I updated the PR, so the focus path should be clean now. Let me know if you see anything else.

Kapture 2022-11-24 at 11 42 38

Note that I keep the dropdown list unfocussable. That's the state beforehand to ensure it works correctly with up-down navigation (similar to the native select dropdown).

**Comment by vsgoulart** [comment]
https://user-images.githubusercontent.com/1022916/203797299-ad056c50-21bf-4595-bb5e-9a19ff956fbe.mp4

@pinussilvestrus Found another issue. If you select one of the radio buttons on the field before and press enter the last pill is deleted

File: packages/form-js-viewer/src/render/components/form-fields/Taglist.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply packages/form-js-viewer/src/render/components/form-fields/Taglist.js:110]: Root cause: that's related to a missing [`type` attribute](https://www.w3schools.com/tags/att_button_type.asp) (default is `submit`).

**Comment by pinussilvestrus** [comment]
I forgot to add the `type` attribute for the remove button. Should work now 👍

File: packages/form-js-viewer/src/render/components/form-fields/Taglist.js [hunk 2]
```diff
@@ -113,22 +114,44 @@ export default function Taglist(props) {
...
-          return (
-            <div class="fjs-taglist-tag" onMouseDown={ (e) => e.preventDefault() }>
-              <span class="fjs-taglist-tag-label">
-                {valueToOptionMap[v] ? valueToOptionMap[v].label : `unexpected value{${v}}`}
-              </span>
-              <span class="fjs-taglist-tag-remove" onMouseDown={ () => deselectValue(v) }><CloseIcon /></span>
-            </div>
-          );
-        })
+        <div class="fjs-taglist-tags">
```
Comment (vsgoulart) [inline packages/form-js-viewer/src/render/components/form-fields/Taglist.js:135]: I'll have to check if this doesn't break the styling on Tasklist. We have to setup a system for changes on components in order to not break customization

**Review by vsgoulart — APPROVED** [review]

File: packages/form-js-viewer/src/render/components/form-fields/Taglist.js (hunk 2 above)
Comment (pinussilvestrus) [reply packages/form-js-viewer/src/render/components/form-fields/Taglist.js:135]: I agree with being careful with this kind of change. I introduced this just to ensure the tags are encapsulated from the search input. I can also revert this if you see any problem in Tasklist 👍

Comment (vsgoulart) [reply packages/form-js-viewer/src/render/components/form-fields/Taglist.js:135]: I just rebased locally with `develop` and tested it on Tasklist. This PR should be good to merge.
I just need to fix the focus state, but it's unrelated to this PR

Comment (pinussilvestrus) [reply packages/form-js-viewer/src/render/components/form-fields/Taglist.js:135]: thanks for the confirmation 👍

**Pull request merged**